compiler/rustc_lexer/src/cursor.rs

```diff
@@ -91,7 +91,17 @@ impl<'a> Cursor<'a> {
         // It was tried making optimized version of this for eg. line comments, but
         // LLVM can inline all of this and compile it down to fast iteration over bytes.
         while predicate(self.first()) && !self.is_eof() {
-            self.bump();
+            #[cfg(debug_assertions)]
+            {
+                self.bump();
```

Comment:
why is this not `unwrap`ed?

Reply:
because it wasn't before? the unwrap_unchecked below is there to tell it the Err result is unreachable